### PR TITLE
Add Expo source maps guide

### PR DESCRIPTION
**Pull request by krystofwoldrich**
Updates the docs to match the latest version of the RN SDK

- https://github.com/getsentry/sentry-react-native/releases/tag/5.16.0-alpha.4

### CODE REVIEW

**Comment by shanamatthews** [comment]
I've started reviewing this and will finish it up tomorrow!

**Comment by krystofwoldrich** [comment]
@shanamatthews Thank you. Just FYI: Only the Expo pages are new, the Hermes and JSC are moved from the main RN source maps page to its own.

File: src/platforms/react-native/sourcemaps/uploading/expo-advanced.mdx [hunk 1]
```diff
@@ -0,0 +1,216 @@
+---
+title: Expo (Advanced)
+description: "Manually upload source maps for native Expo releases."
+---
+
+Sentry's React Native SDK works out of the box with Expo applications. To see readable stack traces in the product source maps need to be uploaded to Sentry. This guide explains how to manually upload source maps for Expo applications releases.
```
Comment (shanamatthews) [inline src/platforms/react-native/sourcemaps/uploading/expo-advanced.mdx:6]:
```suggestion
Sentry's React Native SDK works out of the box with Expo applications. To see readable stack traces in the product, you must upload source maps to Sentry. This guide explains how to manually upload source maps for Expo application releases.
```

File: src/platforms/react-native/sourcemaps/uploading/expo-advanced.mdx [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+
+If you are looking for a guide on how to automatically upload source maps for Expo applications releases or updates, please refer to the following guide:
+
+- [How to upload source maps for Expo releases and EAS Updates](/platforms/react-native/sourcemaps/uploading/expo/)
```
Comment (shanamatthews) [inline src/platforms/react-native/sourcemaps/uploading/expo-advanced.mdx:10]:
```suggestion
For a guide on how to automatically upload source maps for Expo application releases or updates, see the [Expo](/platforms/react-native/sourcemaps/uploading/expo/) source maps docs.
```

File: src/platforms/react-native/sourcemaps/uploading/expo-advanced.mdx [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+- [Sign up for an account](https://sentry.io/signup/)
+- [Set up Sentry React Native SDK](/platforms/react-native/manual-setup/expo/) version 5.16.0-alpha.4 or newer
+
+<OrgAuthTokenNote />
+
+<Include name="react-native-sourcemaps-expo-common-instructions.mdx" />
+
+## Manual Upload for Hermes Release
+
+After the Sentry Expo Plugin and Sentry Metro plugin are added to the application configuration, generate and upload Hermes source maps.
```
Comment (shanamatthews) [inline src/platforms/react-native/sourcemaps/uploading/expo-advanced.mdx:23]:
```suggestion
After the Sentry Expo Plugin and the Sentry Metro plugin are added to the application configuration, generate and upload Hermes source maps.
```

File: src/platforms/react-native/sourcemaps/uploading/expo-advanced.mdx [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+
+### Generate bundle
+
+To generate a bundle and source map for the native application release use the following command:
```
Comment (shanamatthews) [inline src/platforms/react-native/sourcemaps/uploading/expo-advanced.mdx:27]:
```suggestion
To generate a bundle and source maps for the native application release, use the following command:
```

File: src/platforms/react-native/sourcemaps/uploading/expo-advanced.mdx [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+  --bundle-output main.jsbundle \
+  --sourcemap-output main.jsbundle.map \
+  --minify false
+```
+
+#### Notes
+
+- The actual command might differ depending on the application configuration. The above commands are the default commands used by Expo.
+- Bundle and Source Map names can change the generated Debug ID.
+- For custom entry file paths the `--entry-file` parameter needs to be adjusted.
```
Comment (shanamatthews) [inline src/platforms/react-native/sourcemaps/uploading/expo-advanced.mdx:57]:
```suggestion
- The actual command might differ depending on your application configuration. The above commands are the default commands used by Expo.
- Bundle and source maps names can change the generated debug ID.
- For custom entry file paths, the `--entry-file` parameter needs to be adjusted.
```

File: src/platforms/react-native/sourcemaps/uploading/expo-advanced.mdx [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+
+<OrgAuthTokenNote />
+
+<Include name="react-native-sourcemaps-expo-common-instructions.mdx" />
+
+## Manual Upload for Hermes Release
+
+After the Sentry Expo Plugin and Sentry Metro plugin are added to the application configuration, generate and upload Hermes source maps.
+
+### Generate bundle
```
Comment (shanamatthews) [inline src/platforms/react-native/sourcemaps/uploading/expo-advanced.mdx:25]:
```suggestion
### Generate Bundle
```

File: src/platforms/react-native/sourcemaps/uploading/expo-advanced.mdx [hunk 7]
```diff
@@ -0,0 +1,216 @@
...
+  --minify false
+```
+
+#### Notes
+
+- The actual command might differ depending on the application configuration. The above commands are the default commands used by Expo.
+- Bundle and Source Map names can change the generated Debug ID.
+- For custom entry file paths the `--entry-file` parameter needs to be adjusted.
+
+### Compile bytecode
```
Comment (shanamatthews) [inline src/platforms/react-native/sourcemaps/uploading/expo-advanced.mdx:59]:
```suggestion
### Compile Bytecode
```

File: src/platforms/react-native/sourcemaps/uploading/expo-advanced.mdx [hunk 8]
```diff
@@ -0,0 +1,216 @@
...
+
+Compile Hermes bytecode bundle and source map:
```
Comment (shanamatthews) [inline src/platforms/react-native/sourcemaps/uploading/expo-advanced.mdx:61]:
```suggestion
Compile Hermes bytecode bundle and source maps:
```

File: src/platforms/react-native/sourcemaps/uploading/expo-advanced.mdx [hunk 9]
```diff
@@ -0,0 +1,216 @@
...
+  main.jsbundle.packager.map \
+  main.jsbundle.hbc.map \
+  -o main.jsbundle.map
+node \
+  node_modules/@sentry/react-native/scripts/copy-debugid.js \
+  main.jsbundle.packager.map main.jsbundle.map
+rm -f main.jsbundle.packager.map
+```
+
+### Upload source maps
```
Comment (shanamatthews) [inline src/platforms/react-native/sourcemaps/uploading/expo-advanced.mdx:130]:
```suggestion
### Upload Source Maps
```

File: src/platforms/react-native/sourcemaps/uploading/expo-advanced.mdx [hunk 10]
```diff
@@ -0,0 +1,216 @@
...
+
+Make sure `sentry-cli` is configured for your project and set up your environment variables:
+
+```bash {filename:.env}
+SENTRY_ORG=___ORG_SLUG___
+SENTRY_PROJECT=___PROJECT_SLUG___
+SENTRY_AUTH_TOKEN=___ORG_AUTH_TOKEN___
+```
+
+Upload the bundle and source map to Sentry:
```
Comment (shanamatthews) [inline src/platforms/react-native/sourcemaps/uploading/expo-advanced.mdx:140]:
```suggestion
Upload the bundle and source maps to Sentry:
```

File: src/platforms/react-native/sourcemaps/uploading/expo-advanced.mdx [hunk 11]
```diff
@@ -0,0 +1,216 @@
...
+  --debug-id-reference \
+  --strip-prefix /path/to/project/root \
+  main.jsbundle main.jsbundle.map
+```
+
+## Manual Upload for JSC Release
+
+After the Sentry Expo Plugin and Sentry Metro plugin are added to the application configuration, generate and upload Javascript Core source maps.
+
+### Generate bundle
```
Comment (shanamatthews) [inline src/platforms/react-native/sourcemaps/uploading/expo-advanced.mdx:160]:
```suggestion
### Generate Bundle
```

File: src/platforms/react-native/sourcemaps/uploading/expo-advanced.mdx [hunk 12]
```diff
@@ -0,0 +1,216 @@
...
+
+To generate a bundle and source map for the native application release use the following command:
```
Comment (shanamatthews) [inline src/platforms/react-native/sourcemaps/uploading/expo-advanced.mdx:162]:
```suggestion
To generate a bundle and source maps for the native application release, use the following command:
```

File: src/platforms/react-native/sourcemaps/uploading/expo-advanced.mdx [hunk 13]
```diff
@@ -0,0 +1,216 @@
...
+  --bundle-output main.jsbundle \
+  --sourcemap-output main.jsbundle.map \
+  --minify true
+```
+
+#### Notes
+
+- The actual command might differ depending on the application configuration. The above commands are the default commands used by Expo.
+- Bundle and Source Map names can change the generated Debug ID.
+- For custom entry file paths the `--entry-file` parameter needs to be adjusted.
```
Comment (shanamatthews) [inline src/platforms/react-native/sourcemaps/uploading/expo-advanced.mdx:192]:
```suggestion
- The actual command might differ depending on your application configuration. The above commands are the default commands used by Expo.
- Bundle and source maps names can change the generated debug ID.
- For custom entry file paths, the `--entry-file` parameter needs to be adjusted.
```

File: src/platforms/react-native/sourcemaps/uploading/expo-advanced.mdx [hunk 14]
```diff
@@ -0,0 +1,216 @@
...
+
+### Upload source maps
```
Comment (shanamatthews) [inline src/platforms/react-native/sourcemaps/uploading/expo-advanced.mdx:194]:
```suggestion
### Upload Source Maps
```

File: src/platforms/react-native/sourcemaps/uploading/expo.mdx [hunk 15]
```diff
@@ -0,0 +1,53 @@
+---
+title: Expo
+description: "Upload source maps for native Expo releases and EAS Updates."
+---
+
+Sentry's React Native SDK works out of the box with Expo applications. To see readable stack traces in the product source maps need to be uploaded to Sentry. This guide explains how to upload source maps for Expo applications.
```
Comment (shanamatthews) [inline src/platforms/react-native/sourcemaps/uploading/expo.mdx:6]:
```suggestion
Sentry's React Native SDK works out of the box with Expo applications. To see readable stack traces in the product, you must upload source maps to Sentry. This guide explains how to upload source maps for Expo applications.
```

File: src/platforms/react-native/sourcemaps/uploading/expo.mdx [hunk 16]
```diff
@@ -0,0 +1,53 @@
...
+- [Sign up for an account](https://sentry.io/signup/)
+- [Set up Sentry React Native SDK](/platforms/react-native/manual-setup/expo/) version 5.16.0-alpha.4 or newer
+
+<OrgAuthTokenNote />
+
+<Include name="react-native-sourcemaps-expo-common-instructions.mdx" />
+
+## Automatic Upload for Release
+
+When Sentry Expo Plugin `@sentry/react-native/expo` and Sentry Metro Pugin are added to the application configuration source maps for the native builds are uploaded automatically during EAS Builds. And also when building the native application release locally using `expo prebuild`.
```
Comment (shanamatthews) [inline src/platforms/react-native/sourcemaps/uploading/expo.mdx:19]:
```suggestion
When the Sentry Expo Plugin `@sentry/react-native/expo` and the Sentry Metro Pugin are added to your application configuration, source maps for the native builds are uploaded automatically during EAS Builds and when building the native application release locally using `expo prebuild`.
```

File: src/platforms/react-native/sourcemaps/uploading/expo.mdx [hunk 17]
```diff
@@ -0,0 +1,53 @@
...
+
+## Manual Upload for EAS Updates
+
+To upload source maps for EAS Updates and `expo export` set up the Sentry Expo Plugin, Sentry Metro plugin and execute the `sentry-expo upload-sourcemaps` command.
```
Comment (shanamatthews) [inline src/platforms/react-native/sourcemaps/uploading/expo.mdx:23]:
```suggestion
To upload source maps for EAS Updates and `expo export`, set up the Sentry Expo Plugin, the Sentry Metro plugin and execute the `sentry-expo upload-sourcemaps` command.
```

File: src/platforms/react-native/sourcemaps/uploading/expo.mdx [hunk 18]
```diff
@@ -0,0 +1,53 @@
...
+### Create Update
+
+To create an update for the application use the following command:
+
+```bash
+# expo export
+eas update
+```
+
+### Upload source maps
```
Comment (shanamatthews) [inline src/platforms/react-native/sourcemaps/uploading/expo.mdx:34]:
```suggestion
### Upload Source Maps
```

File: src/platforms/react-native/sourcemaps/uploading/expo.mdx [hunk 19]
```diff
@@ -0,0 +1,53 @@
...
+
+#### Notes
+
+- `dist` is the default output directory of `eas update`.
+
+## Notes
+
+If you are looking for a guide on how to manually upload source maps for Expo applications releases, please refer to the following guide:
+
+- [How to manually upload source maps for Expo releases](/platforms/react-native/sourcemaps/uploading/expo-advanced/)
```
Comment (shanamatthews) [inline src/platforms/react-native/sourcemaps/uploading/expo.mdx:53]:
```suggestion
For a guide on how to manually upload source maps for Expo application releases, see the [Expo (Advanced)](/platforms/react-native/sourcemaps/uploading/expo-advanced/) source maps docs.
```

**Review by shanamatthews — APPROVED** [review]
Thanks for pointing out that only the Expo docs are new!

I made suggestions to standardize on "source maps" vs. "source map" because I think that's our convention, but feel free to switch it back if you prefer.

File: src/includes/react-native-sourcemaps-expo-common-instructions.mdx [hunk 20]
```diff
@@ -0,0 +1,15 @@
+## Common Setup
+
+To upload source maps to Sentry Sentry Expo Plugin and Sentry Metro Plugin needs to be add to the Expo application.
+
+### Add the Sentry Expo Plugin
+
+To ensure the bundles and source maps are automatically uploaded during the local and EAS native applications builds, add `@sentry/react-native/expo` to the Expo application configuration:
```
[the remark this answers is not in the document]
Comment (brentvatne) [reply src/includes/react-native-sourcemaps-expo-common-instructions.mdx:7]: > add `@sentry/react-native/expo` to the Expo application configuration:

maybe instead:

> add the `@sentry/react-native/expo` config plugin to the Expo application configuration:

File: src/includes/react-native-sourcemaps-expo-common-instructions.mdx [hunk 21]
```diff
@@ -0,0 +1,15 @@
...
+
+<Include name="react-native-expo-plugin-code-snippet.mdx" />
+
+### Add Sentry Metro Plugin
+
+To ensure unique Debug IDs get assigned to the generated bundles and source maps, add Sentry Serializer to the Metro configuration:
```
[the remark this answers is not in the document]
Comment (brentvatne) [reply src/includes/react-native-sourcemaps-expo-common-instructions.mdx:13]: > To ensure unique Debug IDs get assigned to the generated bundles and source maps, add Sentry Serializer to the Metro configuration:

->

> To ensure unique Debug IDs are assigned to the generated bundles and source maps, add the Sentry Metro Serializer to the Metro configuration:

File: src/platforms/react-native/sourcemaps/uploading/expo.mdx (hunk 15 above)
Comment (brentvatne) [reply src/platforms/react-native/sourcemaps/uploading/expo.mdx:6]: > Upload source maps for native Expo releases and EAS Updates

somewhat confusingly, the library is called `expo-updates` but the service is called "EAS Update"

File: src/platforms/react-native/sourcemaps/uploading/expo.mdx (hunk 17 above)
Comment (brentvatne) [reply src/platforms/react-native/sourcemaps/uploading/expo.mdx:23]: `sentry-expo upload-sourcemaps` -> `sentry-expo-upload-sourcemaps`

**Comment by brentvatne** [comment]
small nit: when referring to expo commands from Expo CLI (eg: `expo export`) we usually prefix them with `npx` to be clear that this is expected, so `expo export` -> `npx expo export`. this is because we invoke the cli through local `node_modules` rather than a global `expo` bin. so we may want to update that here. as for EAS commands (eg: `eas update`), we should not use `npx` (so we are already doing this right in this doc)

File: src/includes/react-native-sourcemaps-expo-common-instructions.mdx (hunk 21 above)
[the remark this answers is not in the document]
Comment (krystofwoldrich) [reply src/includes/react-native-sourcemaps-expo-common-instructions.mdx:13]:
```suggestion
To ensure unique Debug IDs are assigned to the generated bundles and source maps, add the Sentry Metro Plugin to the configuration:
```

File: src/platforms/react-native/manual-setup/expo.mdx [hunk 22]
```diff
@@ -151,3 +89,13 @@ To verify that everything is working as expected, build the `Release` version of
   }}
 />
 ```
+
+## Next Steps
+
+- [Lear how to upload Source Maps for native builds and Expo Updates](/platforms/react-native/sourcemaps/uploading/expo/)
```
[the remark this answers is not in the document]
Comment (krystofwoldrich) [reply src/platforms/react-native/manual-setup/expo.mdx:95]:
```suggestion
- [Learn how to upload source maps for native builds and Expo Updates](/platforms/react-native/sourcemaps/uploading/expo/)
```

[Expo Updates](https://docs.expo.dev/versions/latest/sdk/updates/) was there intentional I wanted to emphasize that the guide includes steps for the Updates too.

File: src/platforms/react-native/sourcemaps/uploading/expo.mdx [hunk 23]
```diff
@@ -0,0 +1,53 @@
+---
+title: Expo
+description: "Upload source maps for native Expo releases and EAS Updates."
```
Comment (krystofwoldrich) [inline src/platforms/react-native/sourcemaps/uploading/expo.mdx:3]:
```suggestion
description: "Upload source maps for native Expo releases and EAS Update."
```

**Pull request merged**